### ISSUE TITLE
Misleading url component name

### ISSUE THREAD

**Issue by smaant** [dakrone/clj-http#309]
`wrap-url` middleware parses a given URL and replaces it with components, produced by `parse-url` function. One of the components is `:uri`, which is based on implementation contains URL's path and this is very misleading. According to [RFC 2396](https://www.ietf.org/rfc/rfc2396.txt) and general convention in the industry, URI is a whole identifier, not just the part of it.

Also, I think it's worth it to keep original URL (or its encoded version) in the request map for further middlewares and handlers.

**Comment by dakrone** [comment]
Yes, I agree this should be changed to reflect what it actually is. We can do this in master (4.x) now since it's a breaking change

**Comment by cjbarre** [comment]
Hey @dakrone I'd like to adopt this one. My PR is attached. I'll handle any feedback from you.
